airflow/contrib/operators/cloudml_operator.py

```diff
@@ -14,16 +14,321 @@
+    :type uri: string
+
+    :param max_worker_count: The maximum number of workers to be used
+        for parallel processing. Defaults to 10 if not specified.
+    :type max_worker_count: int
+
+    :param runtime_version: The Google Cloud ML runtime version to use
+        for batch prediction.
+    :type runtime_version: string
+
+    :param gcp_conn_id: The connection ID used for connection to Google
+        Cloud Platform.
+    :type gcp_conn_id: string
+
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must
+        have doamin-wide delegation enabled.
+    :type delegate_to: string
+
+    Raises:
+        ValueError: if a unique model/version origin cannot be determined.
+    """
+
+    template_fields = [
+        "prediction_job_request",
+    ]
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 job_id,
+                 region,
+                 data_format,
+                 input_paths,
+                 output_path,
+                 model_name=None,
+                 version_name=None,
+                 uri=None,
+                 max_worker_count=None,
+                 runtime_version=None,
+                 gcp_conn_id='google_cloud_default',
+                 delegate_to=None,
+                 *args,
+                 **kwargs):
+        super(CloudMLBatchPredictionOperator, self).__init__(*args, **kwargs)
+
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+
+        try:
+            prediction_input = _create_prediction_input(
+                project_id, region, data_format, input_paths, output_path,
+                model_name, version_name, uri, max_worker_count,
+                runtime_version)
+        except ValueError as e:
+            logging.error(
+                'Cannot create batch prediction job request due to: {}'
+                .format(str(e)))
+            raise
+
+        self.prediction_job_request = {
+            'jobId': job_id,
+            'predictionInput': prediction_input
+        }
+
+    def execute(self, context):
+        hook = CloudMLHook(self.gcp_conn_id, self.delegate_to)
+        job_id = _normalize_cloudml_job_id(
+            self.prediction_job_request['jobId'])
+        try:
+            existing_job = hook.get_job(self.project_id, job_id)
+            logging.info(
+                'Job with job_id {} already exist: {}.'.format(
+                    job_id,
+                    existing_job))
+            finished_prediction_job = hook.wait_for_job_done(
+                self.project_id,
+                job_id)
+        except errors.HttpError as e:
+            if e.resp.status == 404:
+                logging.error(
+                    'Job with job_id {} does not exist. Will create it.'
+                    .format(job_id))
+                finished_prediction_job = hook.create_job(
```

Comment:
Sorry for the confusing interface. It's not needed as create_job is calling wait_for_job_done under the cover. 
To make it clearer we refactored the hook interface and how the operator uses it. Now only create_job (which creates and wait until a job is finished) is "exposed" and used by the operator.